packages/components-angular/src/app/examples/pin-code-example-controlled.component.ts

```diff
@@ -0,0 +1,27 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import type { PinCodeUpdateEvent } from '@porsche-design-system/components-angular';
+
+@Component({
+  selector: 'page-pin-code-example-controlled',
+  template: `
+    <p-pin-code
+      [label]="'Some Label'"
+      [length]="length"
+      [value]="currentValue"
+      (update)="onUpdate($event)"
+    ></p-pin-code>
+    <p-text>Current value: {{ currentValue }}</p-text>
+    <p-text>Completely filled: {{ isComplete }}</p-text>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PinCodeExampleControlledComponent {
+  length = 4;
+  currentValue = '';
+  isComplete = false;
```

Comment:
Could also write isComplete as a computed property from currentValue:
```suggestion
  isComplete = this.currentValue.length === this.length;
```